packages/components/src/presentation-components/table/UsePresentationTable.ts

```diff
@@ -84,10 +96,59 @@ export function usePresentationTable<TColumn, TRow>(props: UsePresentationTableP
  */
 export function usePresentationTableWithUnifiedSelection<TColumn, TRow>(
   props: Omit<UsePresentationTableProps<TColumn, TRow>, "keys">,
-): UsePresentationTableResult<TColumn, TRow> {
+): UsePresentationTableWithUnifiedSelectionResult<TColumn, TRow> {
   const unifiedSelection = useUnifiedSelectionContext();
   const keys = unifiedSelection?.getSelection() ?? emptyKeySet;
-  return usePresentationTable({ ...props, keys });
-}
+  const [selectedRows, setSelectedRows] = useState<TableRowDefinition[]>();
+
+  const { imodel, ruleset, pageSize, columnMapper, rowMapper } = props;
+  const columns = useColumns({ imodel, ruleset, keys });
+  const { options, sort, filter } = useTableOptions({ columns });
+  const { rows, isLoading, loadMoreRows } = useRows({ imodel, ruleset, keys, pageSize, options });
+
+  const tableGuid: string = useMemo(() => window.crypto.randomUUID(), []);
+
+  useEffect(() => {
+    const disposeListener = Presentation.selection.selectionChange.addListener(({ level, keys: toggledRowKeys, source }) => {
+      if (level !== (unifiedSelection?.selectionLevel ?? 0) + 1 || source === tableGuid) {
+        return;
+      }
+
+      const rowsToAddToSelection: TableRowDefinition[] = [];
+      toggledRowKeys.forEach((key) => {
+        // should return just one row
+        const selectedRow = rows.filter((row) => row.key === JSON.stringify(key))[0];
+        rowsToAddToSelection.push(selectedRow);
+      });
 
+      setSelectedRows(rowsToAddToSelection);
+    });
+
+    return disposeListener;
+  }, [rows, tableGuid, unifiedSelection?.selectionLevel]);
+
+  const onSelect = (selectedData: string[]) => {
```

Comment:
I think this function should not update selection if valid keys are passed but this table does not have rows for those keys.

Reply:
Fixed this by checking if a row for the key exists before adding it to the unifiedSelection.